docs/dev-guide/3.0-development/rest-api.rst

```diff
@@ -0,0 +1,540 @@
+REST API Guidelines
+===================
+
+Introduction
+------------
+
+The Pulp 3 API is intended to be decoupled from the data model, which the Django REST Framework
+(DRF) makes it pretty easy to do. Where needed, support classes have been added to help the REST
+API correctly and consistently represent our specialized models and relationships (most notably
+the Master/Detail relationship).
+
+Our API starts at a :term:`DRF` :term:`Router`. Each :term:`ViewSet` is attached to this
+router, and the router's routes are exposed in urls.py. Subclasses of
+:class:`pulp.app.viewsets.NamedModelViewSet` are automatically registered with the API router,
+and most (possibly all) ViewSets created by plugins should be subclasses of this base class.
+NamedModelViewSets are associated with a Django queryset, a :term:`Serializer` that is able to
+represent members of the Django queryset in the API, and an endpoint name used when registering
+the ViewSet with the API router.
+
+All models exposed via the API must have a corresponding Serializer. Each ViewSet be related to
+a ViewSet, a Since the serializers are so closely related to the models, the serializers and
```

Comment:
Ah, I see what happened. I must've accidentally pasted a thing right in the middle of another thing.